openml/tasks/functions.py

```diff
@@ -236,7 +236,7 @@ def get_task(task_id):
 
     # TODO look into either adding the class labels to task xml, or other
     # way of reading it.
-    class_labels = dataset._retrieve_class_labels()
+    class_labels = dataset._retrieve_class_labels(task.target_feature)
```

Comment:
also, it's weird that the task needs to access a private function of the dataset. That's not good.